testplan/common/remote/remote_resource.py

```diff
@@ -784,3 +811,15 @@ def _transfer_data(
                 stdout=devnull,
                 logger=self.logger,
             )
+
+    def _check_remote_os(self):
```

Comment:
we may store `/etc/os-release` output in `self._remote_os_release` instead, thus no need to make any assumption